pkg/execution/debounce/lua/start.lua

```diff
@@ -0,0 +1,24 @@
+--[[
+--  Updates the debounce pointer to something else
+--  on function start so it doesn't rely on the existing
+--  one on new events
+--
+--  Return value:
+--    0: untouched
+--    1: updated
+-- ]]
+
+local debouncePointerKey = KEYS[1]
+
+local newDebounceID = ARGV[1]
+local existingDebounceID = ARGV[2]
+
+local currentID = redis.call("GET", debouncePointerKey)
+
+-- update the pointer key value only if the existing one matches
+if currentID ~= nil and currentID ~= false and currentID == existingDebounceID then
```

Comment:
Added a condition here to only limit the pointer update if the current debounce ID has not changed yet.